microprofile.ls/org.eclipse.lsp4mp.ls/src/test/java/org/eclipse/lsp4mp/services/ApplicationPropertyExpressionCompletionTest.java

```diff
@@ -0,0 +1,274 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4mp.services;
+
+import static org.eclipse.lsp4mp.services.MicroProfileAssert.c;
+import static org.eclipse.lsp4mp.services.MicroProfileAssert.r;
+import static org.eclipse.lsp4mp.services.MicroProfileAssert.testCompletionFor;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import org.eclipse.lsp4mp.commons.MicroProfileProjectInfo;
+import org.eclipse.lsp4mp.commons.metadata.ItemMetadata;
+import org.eclipse.lsp4mp.ls.commons.BadLocationException;
+import org.junit.Test;
+
+public class ApplicationPropertyExpressionCompletionTest {
+
+	// Tests for where property expression completion should open
+
+	@Test
+	public void justDollarSignNoNewline() throws BadLocationException {
+		String text = //
+				"test.property = hello\n" + //
+						"other.test.property = $|";
+		testCompletionFor(text, generateInfoFor("test.property", "other.test.property"),
+				c("${test.property}", "${test.property}", r(1, 22, 1, 23)));
```

Comment:
I think it's worth defining a new method in `MicroProfileAssert`:

```
	public static CompletionItem c(String text Range range) {
		return c(text, text, range);
	}
```
And using this new method instead of doing `c("${test.property}", "${test.property}", r(1, 22, 1, 23)));`

Reply:
Instead of `r(1, 22, 1, 23)`, can we change it to `r(1, 22, 23)` throughout the PR?